### PR TITLE
Export GRAPHQL_MAX_INT and GRAPHQL_MIN_INT

### LINKED ISSUE #2524: Add max int to exports
# Feature request

GraphQL has a limit to use 32-bit max and min Integer, but js, not built-in 32-bit max Integer.
I prefer not to set constants in the project, if they already defined.
https://github.com/graphql/graphql-js/blob/f31f5050644c3d2925557be3b285032345e91920/src/type/scalars.js#L21-L22

Please, add exports to contents

### PR DESCRIPTION
**Pull request by tofran**
Closes #2524

(Replaces abandoned #2582)

### CODE REVIEW

**Review by saihaj — APPROVED** [review]

File: src/type/scalars.ts [hunk 1]
```diff
@@ -14,8 +14,8 @@ import { GraphQLScalarType } from './definition';
 //
 // n.b. JavaScript's integers are safe between -(2^53 - 1) and 2^53 - 1 because
 // they are internally represented as IEEE 754 doubles.
-const MAX_INT = 2147483647;
-const MIN_INT = -2147483648;
+export const MAX_INT = 2147483647;
```
Comment (IvanGoncharov) [inline src/type/scalars.ts:17]: Please convert above comment to JSDoc
If you have time please write a separate docs for both min and max

Comment (IvanGoncharov) [reply src/type/scalars.ts:17]: Also since we making it public please rename it to `GRAPHQL_INT_MAX` and `GRAPHQL_INT_MIN`.

**Review by IvanGoncharov — CHANGES_REQUESTED** [review]
Please export it in `src/type/index.ts` and `src/index.ts`.
Also please check tests folder for the places where this number (or +-1) is used and replace them with this constant.

**Comment by tofran** [comment]
Hello @IvanGoncharov , thank you for bringing these in and your detailed comment.
I have made the requested changes:

- [x] Checked the tests folder for the places where this number (or +-1) is used and replace them with this constant.*

```sh
$ grep -R "214748364" src  # Intentionally missing the last character
src/type/scalars.ts:export const GRAPHQL_MAX_INT = 2147483647;
src/type/scalars.ts:export const GRAPHQL_MIN_INT = -2147483648;
$ grep -R "2^31" src
src/type/scalars.ts:    'The `Int` scalar type represents non-fractional signed whole numeric values. Int can represent values between -(2^31) and 2^31 - 1.',
$ grep -R "2 \*\* 31" src

```

- [x] Converted above comment to JSDoc - separated and simplified
- [x] Exported it in src/type/index.ts and src/index.ts.
- [x] Renamed to `GRAPHQL_INT_MAX` and `GRAPHQL_INT_MIN`.

---

*Were you also suggesting that tests to these boundaries should be changed?
Ex:
```js
expect(() => parseLiteral('-9876504321')).to.throw(
  'Int cannot represent non 32-bit signed integer value: -9876504321',
);
```
Would now be
```js
expect(() => parseLiteral(GRAPHQL_MIN_INT-1)).to.throw(
  `Int cannot represent non 32-bit signed integer value: ${GRAPHQL_MIN_INT-1}`,
);
```

Should I make this change? Thank you.

**Review by saihaj — APPROVED** [review]

File: src/index.ts [hunk 2]
```diff
@@ -61,6 +61,9 @@ export {
   TypeKind,
   /** Constant Deprecation Reason */
   DEFAULT_DEPRECATION_REASON,
+  /** Int boundaries constants */
+  GRAPHQL_MAX_INT,
+  GRAPHQL_MIN_INT,
```
Comment (IvanGoncharov) [inline src/index.ts:66]: Please match the order in `types/index.ts`

Comment (IvanGoncharov) [reply src/index.ts:66]: Everything else looks good.
Happy to merge once fixed.

**Comment by tofran** [comment]
@IvanGoncharov @saihaj thank you.
Just pushed the requested changes, ready to merge.

**Review by saihaj — APPROVED** [review]

**Comment by tofran** [comment]
@IvanGoncharov I don't think we will be able to merge until you re-review.
Thanks in advance.

**Comment by IvanGoncharov** [comment]
@tofran Thanks 👍 
Merged 🎉

File: src/type/scalars.ts (hunk 1 above)
Comment (jen801) [reply src/type/scalars.ts]: 👍🏻

**Review by jen801 — COMMENTED** [review]
#3355

**Review by Nexuscompute — APPROVED** [review]

**Comment by saihaj** [comment]
Just to avoid noise from potential bot accounts.
